intro_to_rl.ipynb

```diff
@@ -0,0 +1,3225 @@
+{
```

Comment:
<ul><li>I find this language rather too technical for someone seeing this for the first time. Rather than abstract discussions around terminal states, maybe bring this back to the only RL example lots of these tutorial takers know about: the cartpole. We want to maximize the time it stays up, and it gets one reward per timestep. So...</li><li>Also, I would just be confused by a lot of this: why would we want discount factors with infinite horizons etc etc (seems completely arbitrary). Is this really necessary for the first few RL methods? (random policy search doesn't care about discount factors)</li><li>expected discounted return: isn't an expectation something over a distribution? Where have you introduced a distribution, and wouldn't this be E[R_t]?</li></ul><p> </p><p>I think this section is very confusing and would remove the whole thing, or move it to after actually solving the problem (abstractions make more sense then to learners).</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2022/pull/6/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Reply:
<p>I agree with @sbodenstein on this. I think the timing of introducing it should be after an initial model is trained, and we now want to show more things that can be done or are done.</p>

Reply:
<p>I get what you guys are saying @<a href="https://github.com/sbodenstein" target="_blank">sbodenstein</a>&nbsp;and @<a href="https://github.com/ruanvdmerwe" target="_blank">ruanvdmerwe</a> .&nbsp;I think adding a concrete example, i.e. "for CartPole, episode return means we want to maximize the time it stays up..." is a great idea! </p><p> </p><p>It is important to understand Episode Return vs Reward since that is quite foundational to RL, but maybe we move the discounted return part to the beginning of Section 3 (Policy Gradients) as we get more formal. What do you think @<a href="https://github.com/jcformanek" target="_blank">jcformanek</a>&nbsp;? </p><p> </p><p>@<a href="https://github.com/sbodenstein" target="_blank">sbodenstein</a> Why wouldn't you want discount factors with infinite horizons? You typically use a discount factor for infinite horizons. (Not sure if that was a hypothetical question as potential prac attendee).</p><p> </p>

Reply:
<p>I have simplified this section and decided to omit the discussion about discount factors.</p>